src/titiler/xarray/titiler/xarray/extensions.py

```diff
@@ -36,3 +37,55 @@ def variables(
             """return available variables."""
             with self.dataset_opener(src_path, **io_params.as_dict()) as ds:
                 return list(ds.data_vars)  # type: ignore
+
+
+@define
+class DatasetMetadataExtension(FactoryExtension):
+    """Add dataset metadata endpoints to a Xarray TilerFactory."""
+
+    io_dependency: Type[DefaultDependency] = XarrayIOParams
+    dataset_opener: Callable[..., xarray.Dataset] = xarray_open_dataset
+
+    def register(self, factory: TilerFactory):
+        """Register endpoint to the tiler factory."""
+
+        @factory.router.get(
+            "/keys",
```

Comment:
We have called this `/variables` in titiler-multidim
https://github.com/developmentseed/titiler-multidim/blob/5abbf861db9e383a239456f208251fd99ca2196e/src/titiler/multidim/factory.py#L39-L53

I'm not sure which is better, just noting the difference.

Reply:
the `/variables` endpoint is still available in the `VariablesExtension` extension  https://github.com/developmentseed/titiler/blob/57b44d6f2fc32153705550643eb61913f524935f/src/titiler/xarray/titiler/xarray/extensions.py#L16-L38 which will should be deprecated 

https://github.com/xpublish-community/xpublish/blob/main/xpublish/plugins/included/dataset_info.py